File: internal/models/legal_document.go
package models

import (
	"time"

	"github.com/emilijan-koteski/monexa/internal/models/types"
	"gorm.io/gorm"
)

type LegalDocument struct {
	ID                uint               `gorm:"primaryKey" json:"id"`
	CreatedAt         time.Time          `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt         *time.Time         `gorm:"autoUpdateTime" json:"updatedAt"`
	DeletedAt         gorm.DeletedAt     `gorm:"index" json:"-"`
	Type              types.DocumentType `gorm:"not null" json:"type"`
	Version           int                `gorm:"not null" json:"version"`
	Title             string             `gorm:"not null" json:"title"`
	Content           string             `gorm:"not null" json:"content"`
	EffectiveAt       time.Time          `gorm:"not null" json:"effectiveAt"`
	IsActive          bool               `gorm:"not null;default:false" json:"isActive"`
	RequiresReconsent bool               `gorm:"not null;default:true" json:"requiresReconsent"`
}

File: frontend/src/utils/legalDocument.ts
import { Language } from '../enums/Language';
import type { LegalDocument } from '../types/models';

export function getLocalizedTitle(doc: LegalDocument, language: string): string {
  return language === Language.MK && doc.titleMk ? doc.titleMk : doc.title;
}

export function getLocalizedContent(doc: LegalDocument, language: string): string {
  return language === Language.MK && doc.contentMk ? doc.contentMk : doc.content;
}
